### PR TITLE
Dt 858 reference data additional identifiers

### PR DESCRIPTION
**Pull request by andymarke**
- Endpoint to retrieve reference data for a specific data set
- Endpoint to retrieve full list of available data sets

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/ReferenceDataResource.java [hunk 1]
```diff
@@ -45,10 +46,8 @@
             })
     @GetMapping("/probationAreas")
     public Page<KeyValue> getProbationAreaCodes(
-            @ApiParam(name = "active", value = "Restricts to active areas only", example = "true")
-            final @RequestParam(name = "active", required = false) boolean restrictActive,
-            @ApiParam(name = "excludeEstablishments", value = "Restricts to areas that are providers, no prisons will be returned", example = "true")
-            final @RequestParam(name = "excludeEstablishments", required = false) boolean excludeEstablishments) {
+            @ApiParam(name = "active", value = "Restricts to active areas only", example = "true") final @RequestParam(name = "active", required = false) boolean restrictActive,
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/ReferenceDataResource.java:49]: formatting only, no change

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/ReferenceDataResource.java [hunk 2]
```diff
@@ -68,8 +67,7 @@
             })
     @GetMapping(value = "/probationAreas/code/{code}/localDeliveryUnits")
     public Page<KeyValue> getLdusForProbationCode(
-            @ApiParam(name = "code", value = "Probation area code", example = "NO2", required = true)
-            final @PathVariable String code) {
+            @ApiParam(name = "code", value = "Probation area code", example = "NO2", required = true) final @PathVariable String code) {
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/ReferenceDataResource.java:70]: formatting only, no change

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/ReferenceDataResource.java [hunk 3]
```diff
@@ -89,12 +87,47 @@
             })
     @GetMapping(value = "/probationAreas/code/{code}/localDeliveryUnits/code/{lduCode}/teams")
     public Page<KeyValue> getTeamsForLdu(
-            @ApiParam(name = "code", value = "Probation area code", example = "NO2", required = true)
-            final @PathVariable String code,
-            @ApiParam(name = "lduCode", value = "Local delivery unit code", example = "NO2NPSA", required = true)
-            final @PathVariable String lduCode) {
+            @ApiParam(name = "code", value = "Probation area code", example = "NO2", required = true) final @PathVariable String code,
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/ReferenceDataResource.java:90]: formatting only, no change

File: src/main/java/uk/gov/justice/digital/delius/jwt/JwtValidator.java [hunk 4]
```diff
@@ -32,8 +31,6 @@ public void validateJwt(JoinPoint joinPoint) {
 
         ArrayList<Object> argsList = Lists.newArrayList(joinPoint.getArgs());
 
-        log.info("Received call to {} with arguments {}", joinPoint.getSignature().getName(), argsList.stream().map(Object::toString).collect(Collectors.joining(", ")));
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/jwt/JwtValidator.java:35]: Bearer token for the non secure endpoints were being logged

File: src/main/java/uk/gov/justice/digital/delius/service/ReferenceDataService.java [hunk 5]
```diff
@@ -147,4 +151,31 @@ private StandardReference getCustodyEventTypeFor(String code) {
                 .flatMap(borough -> borough.getDistricts().stream())
                 .filter(district -> ynToBoolean(district.getSelectable()));
     }
+
+    public Optional<List<ReferenceData>> getReferenceDataForSet(String set) {
+        return referenceDataMasterRepository
+                .findByCodeSetName(set)
+                .map(referenceDataMaster -> referenceDataMaster.getStandardReferences()
```
Comment (andymarke) [inline src/main/java/uk/gov/justice/digital/delius/service/ReferenceDataService.java:158]: I will move this to a `Transformer` class

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/ReferenceDataResource.java [hunk 6]
```diff
@@ -89,12 +87,47 @@
...
+    @ApiOperation(value = "Return the reference data items for the given set")
+    @ApiResponses(
+            value = {
+                    @ApiResponse(code = 400, message = "Invalid request", response = ErrorResponse.class),
+                    @ApiResponse(code = 401, message = "Unauthorised", response = ErrorResponse.class),
+                    @ApiResponse(code = 403, message = "Forbidden", response = ErrorResponse.class),
+                    @ApiResponse(code = 404, message = "Data set not found", response = ErrorResponse.class),
+                    @ApiResponse(code = 500, message = "Unrecoverable error whilst processing request.", response = ErrorResponse.class)
+            })
+    @GetMapping(value = "/referenceData/set/{set}")
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/delius/controller/secure/ReferenceDataResource.java:106]: should really be `reference-data` but doesn't fit with rest of community api

File: src/main/java/uk/gov/justice/digital/delius/jwt/JwtValidator.java (hunk 4 above)
Comment (petergphillips) [reply src/main/java/uk/gov/justice/digital/delius/jwt/JwtValidator.java:35]: :+1:

File: .circleci/config.yml [hunk 7]
```diff
@@ -5,7 +5,7 @@ orbs:
   hmpps: ministryofjustice/hmpps@1.0
 
 jobs:
-  validate:
+  validate-unit:
```
Comment (andymarke) [inline .circleci/config.yml:8]: Get circle to run unit and integration tests in parallel

File: src/main/java/uk/gov/justice/digital/delius/controller/secure/ReferenceDataResource.java (hunk 6 above)
Comment (andymarke) [reply src/main/java/uk/gov/justice/digital/delius/controller/secure/ReferenceDataResource.java:106]: yeah agreed, thought it best to keep with tradition

**Pull request merged**